Project 7 - Bankist/js/script.js

```diff
@@ -0,0 +1,515 @@
+        avatar: "user.png",
+        transaction_amount: 1000,
+        language: "pt-PT",
+        currency: "EUR",
+        date: "2023-10-01T06:38:17.037Z",
+      },
+    ],
+    loans: [
+      {
+        interest: 6000,
+        loanType: "home",
+        principal: 10000,
+        rate: 0.1,
+        time: 6,
+        totalAmount: 16000,
+        date: "2023-10-09T09:05:50.482Z",
+      },
+    ],
+  },
+  {
+    account_id: 2,
+    name: "Amazon",
+    avatar: "amazon.png",
+    total_amount: 0,
+    interest_earned: 300,
+    currency: "USD",
+    language: "US",
+    interest_rate: 6,
+    pin: "2222",
+    transactions: [
+      {
+        account_id: 1,
+        name: "Sanjay Sah",
+        avatar: "sanjay.jpg",
+        transaction_amount: 100,
+        language: "US",
+        currency: "USD",
+        date: "2023-10-01T06:38:17.037Z",
+      },
+    ],
+  },
+  {
+    account_id: 3,
+    name: "Flipkart",
+    avatar: "flipkart.png",
+    total_amount: 0,
+    interest_earned: 200,
+    currency: "INR",
+    language: "HI",
+    interest_rate: 6,
+    pin: "3333",
+    transactions: [
+      {
+        account_id: 1,
+        name: "Sanjay Sah",
+        avatar: "sanjay.jpg",
+        transaction_amount: 200,
+        currency: "INR",
+        language: "HI",
+        date: "2023-10-08T06:38:17.037Z",
+      },
+    ],
+  },
+];
+
+const datalistUsers = document.getElementById("users-list");
+const datalistReceiver = document.getElementById("receiver-name-list");
+const switchUsersList = document.querySelector(".switch-users-list");
+
+const inputUserName = document.getElementById("user-id");
+const inputUserPin = document.getElementById("user-pin");
+const inputReceiverName = document.getElementById("receiver-name");
+const inputAmountTransfer = document.getElementById("amount-transfer");
+const inputAmountLoan = document.getElementById("amount-loan");
+const inputSelectLoanType = document.getElementById("loan-type");
+
+const btnSort = document.getElementById("sort");
+
+const containerLogin = document.querySelector(".login__container");
+const containerApp = document.querySelector(".main__app");
+
+const labelStatusMessage = document.querySelector(".status__message");
+const labelWelcome = document.querySelector(".welcome");
+const labelCurrentBalance = document.getElementById("current-balance");
+const labelInterestEarned = document.getElementById("interest-earned");
+const labelTotalDeposit = document.getElementById("total__deposit");
+const labelTotalWithdrawn = document.getElementById("total__withdrawn");
+const labelLoanAmount = document.getElementById("loan__amount");
+
+const formLogin = document.getElementById("form-login");
+const formTransfer = document.getElementById("form-transfer");
+const formLoan = document.getElementById("form-loan");
+
+const imgLoggedInUserAvatar = document.getElementById("current-user-avatar");
+
+const transactionLists = document.getElementById("transaction__lists");
+
+let loggedInUser;
```

Comment:
@sknepali Always give initial value to variable if possible.
It's currently undefined.
And Its used in many places without any null check

Reply:
Sir, When the page loads very first time, user will see login form. The logic to show login page and main app depends on. loggedInUser global variable. Initially this is undefined so, the login form will be displayed, when the user click on signin button, we are simply setting the this variable to the current logged in user object. And rest of the method will be access only if the main app will be displayed.